processor/elasticinframetricsprocessor/processor_test.go

```diff
@@ -0,0 +1,95 @@
+package elasticinframetricsprocessor
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/processor"
+	"go.uber.org/zap"
+)
+
+func TestProcessMetrics(t *testing.T) {
+	testCases := []struct {
+		name               string
+		cfg                *Config
+		createMetrics      func() pmetric.Metrics
+		expectedHostname   string
+		expectedMetricName string
+		expetedMetricValue int64
+	}{
+		{
+			name: "ProcessMetrics when AddSystemMetrics is enabled",
```

Comment:
Do we rely on the `opentelemetry-lib` tests? Should we verify a hostmetric remap?  E.g. that `"system.cpu.total.pct"` is added

Reply:
I think the remapping-specific logic is being tested in the opentelemetry-lib. 
So that should be sufficient.